moments/Parsing.py

```diff
@@ -0,0 +1,1149 @@
+        given, allows low-confidence ancestral state assignments- represented 
+        by lower-case nucleotide codes- to stand. If False, sites with them
+        are skipped.
+    :type skip_low_conf: bool, optional
+    :param use_AA: If True, use entries in the VCF field ``INFO/AA`` to polarize
+        alleles (default False). 
+    :type use_AA: bool, optional
+    :param filters: A dictionary mapping VCF fields to filter criteria, for
+        imposing quantitative thresholds on measures of genotype quality and 
+        subsetting to qualitative classes of sites. Filtering is discussed
+        above.
+    :type filters: dict, optional
+    :param allow_multiallelic: If True (default False), includes sites with 
+        more than one alternate allele, counting each derived allele at such 
+        sites as a separate entry in the SFS- otherwise multiallelic sites are
+        skipped. When True, also allows biallelic sites where neither alternate 
+        nor reference alleles match the ancestral state to be counted- these
+        are also otherwise skipped.
+    :type allow_multiallelic: bool, optional
+    :param sample_sizes: Dictionary mapping populations to sample sizes 
+        (default None). The output SFS will be projected to these sample sizes.
+        If not given, it will default to the sample sizes implied by `ploidy`
+        and the number of individuals in each population.
+    :type sample_sizes: dict, optional
+    :param mask_corners:
+    :type mask_corners: bool, optional
+    :param ploidy: Optional (default 2), defines the maximum derived allele 
+        count in combination with the number of samples.
+    :type plody: int, optional
+    :param verbose: If 0, print only warning messages- if 1, print soft error
+        messages the first time they are triggered- if an integer larger than 1,
+        do the same and print a progress message every `verbose` lines 
+        (default 1)
+    :type verbose: int, optional
+    :param folded: If True, return the folded SFS (default False).
+    :type folded: bool, optional
+   
+    :returns: The SFS, represented as a ``moments.Spectrum`` instance.
+    :rtype: moments.Spectrum
+    """
+    data = _tally_vcf(
+        vcf_file,
+        pop_mapping=pop_mapping,
+        pop_file=pop_file,
+        bed_file=bed_file,
+        interval=interval,
+        anc_seq_file=anc_seq_file,
+        allow_low_confidence=allow_low_confidence,
+        use_AA=use_AA,
+        filters=filters,
+        allow_multiallelic=allow_multiallelic,
+        verbose=verbose,
+        ploidy=ploidy
+    )
+    fs = _spectrum_from_tally(
+        data, 
+        sample_sizes=sample_sizes, 
+        mask_corners=mask_corners
+    )
+    if folded:
+        fs = fs.fold()
+
+    return fs
+
+
+stats = defaultdict(int)
+
+
+def _clear_stats():
+    """
+    Re-initialize `stats`.
+    """
+    global stats
+    stats = defaultdict(int)
+    return
+
+
+def current_time():
```

Comment:
This, print_err and print_out might want to be private functions? If not underscored, they will show up in the API documentation, and it might get a bit cluttered.